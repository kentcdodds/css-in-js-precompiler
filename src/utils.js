// eslint-disable-next-line import/prefer-default-export
export {looksLike}

// generic utils
function looksLike(a, ...bs) {
  return (
    a &&
    bs.length &&
    bs.some(b =>
      Object.keys(b).every(bKey => {
        const bVal = b[bKey]
        const aVal = a[bKey]
        if (typeof bVal === 'function') {
          return bVal(aVal)
        }
        return isPrimitive(bVal) ? bVal === aVal : looksLike(aVal, bVal)
      }),
    )
  )
}

function isPrimitive(val) {
  // eslint-disable-next-line
  return val == null || /^[sbn]/.test(typeof val);
}
