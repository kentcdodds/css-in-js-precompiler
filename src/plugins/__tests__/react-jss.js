import fs from 'fs'
import * as recast from 'recast'
import {formatOutput, fixturePath} from './helpers/utils'

const reactJssPlugin = require('../react-jss')
const precompile = require('../../')

const babelOptions = {
  parserOpts: {parser: recast.parse},
  generatorOpts: {generator: recast.print, lineTerminator: '\n'},
}

const tests = [
  {
    modifier: 'skip',
    title: 'basic',
    fixtureName: 'basic.js',
  },
]

tests.forEach(({title, fixtureName, modifier}, index) => {
  if (modifier) {
    test[modifier](title, testFn)
  } else {
    test(title, testFn)
  }
  function testFn() {
    const filename = fixturePath(`jss/${fixtureName}`)
    const sourceCode = fs.readFileSync(filename, 'utf8')
    const {transformed, css} = precompile({
      plugin: reactJssPlugin,
      sources: [{filename, code: sourceCode, babelOptions}],
    })

    const [{code}] = transformed
    const output = formatOutput({sourceCode, code, css})

    expect(code).not.toBe(sourceCode)
    expect(css).toBeTruthy()
    expect(output).toMatchSnapshot(`${index + 1}. ${title}`)
  }
})
