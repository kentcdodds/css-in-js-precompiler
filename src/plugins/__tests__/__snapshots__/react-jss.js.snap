// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. basic 1`] = `
"
import React from 'react'
import injectSheet from 'react-jss'

const styles = {
  button: {
    fontWeight: 'bold',
  },
}

const Button = ({classes, children}) =>
  React.createElement('button', {className: classes.button})

export default injectSheet(styles)(Button)

    ðŸ‘‡

import React from 'react'
import injectSheet from 'react-jss'

const styles = {
  button: \\"button-0-0\\"
}

const Button = ({classes, children}) =>
  React.createElement('button', {className: classes.button})

export default injectSheet(styles)(Button)

.button-0-0 {
  font-weight: bold;
}
"
`;
